core/src/main/scala/org/apache/spark/deploy/history/HistoryServerArguments.scala

```diff
@@ -48,7 +48,11 @@ private[history] class HistoryServerArguments(conf: SparkConf, args: Array[Strin
       case Nil =>
 
       case _ =>
-        printUsageAndExit(1)
+        logWarning("Setting log directory through the command line is deprecated as of " +
+          "Spark 1.1.0. Please set this through spark.history.fs.logDirectory instead.")
+        conf.set("spark.history.fs.logDirectory", args.head)
```

Comment:
Why is this necessary? Looking at the current script, no option is passed to the history server at all. Anybody who worked around the script needed to use the `-d` argument. So this is basically removing the check for invalid command lines, which can lead to confusing behavior (e.g. `java HistoryServer blah -d directory` will set the directory to `blah` with your code, instead of telling the user his command line arguments make no sense).